### PR TITLE
Add User Tool as LTI Tool

### PR DESCRIPTION
**Pull request by conradbeach**
Pivotal Tracker: [#171524038](https://www.pivotaltracker.com/story/show/171524038)

This adds the user tool as an LTI tool to Adhesion.

It's adds a shell of the user tool client app and adds the user tool to the list of applications in `seeds.rb`.

The tool is configured to display in the account navigation for account admins.

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -384,6 +384,45 @@
...
+      default: [],
+      common: [
+        "urn:lti:sysrole:ims/lis/SysAdmin",
+        "urn:lti:sysrole:ims/lis/Administrator",
+        "urn:lti:instrole:ims/lis/Administrator",
+      ],
+      GET_SUB_ACCOUNTS_OF_ACCOUNT: [],
+      LIST_USERS_IN_ACCOUNT: [],
+      EDIT_USER: [],
+    },
```
Comment (conradbeach) [inline db/seeds.rb]: The values here in `canvas_api_permissions` are somewhat of a guess.

File: client/apps/user_tool/components/layout/index.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  let result;
+  let props;
+
+  beforeEach(() => {
+    props = {};
+    result = ReactTestUtils.renderIntoDocument(<Index {...props} />);
+  });
+
+  it('renders the index', () => {
+    expect(ReactDOM.findDOMNode(result)).toBeDefined();
```
[the remark this answers is not in the document]
Comment (conradbeach) [reply client/apps/user_tool/components/layout/index.spec.jsx]: This is how it's done in the other app I copied from. I tried the alternative implementation suggested in the Eslint docs, but I failed to get it to work. What would be a good alternative here?

[the remark this answers is not in the document]
Comment (srphillips) [reply client/apps/user_tool/components/layout/index.spec.jsx]: I think this is an exception we'll have to make seeing as how it is handled the same in the other apps.  Perhaps @bfcoder has some thoughts.

File: db/seeds.rb (hunk 1 above)
Comment (srphillips) [reply db/seeds.rb]: I think this has most everything, but let's also include editing logins, and searching for users.

File: client/apps/user_tool/components/layout/index.spec.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (bfcoder) [reply client/apps/user_tool/components/layout/index.spec.jsx]: I'm fine making this exception, as this is how we have been doing it.

File: app/models/application.rb [hunk 3]
```diff
@@ -36,6 +36,7 @@ class Application < ApplicationRecord
   SURVEYAGGREGATION = "surveys".freeze
   POSTGRADES = "postgrades".freeze
   COURSECOMPLETION = "coursecompletion".freeze
+  USERTOOL = "usertool".freeze
```
Comment (srphillips) [inline app/models/application.rb]: @bfcoder does this also need to match for sub-domains?

File: db/seeds.rb (hunk 1 above)
Comment (srphillips) [reply db/seeds.rb]: Or at least the logins stuff.  Searching for users should be based on parameters we can pass to the `LIST_USERS_IN_ACCOUNT`

File: db/seeds.rb [hunk 4]
```diff
@@ -384,6 +384,45 @@
...
+      title: "User Tool",
+      description: "User Tool Application",
+      privacy_level: "public",
+      icon: "oauth_icon.png",
+      custom_fields: {
+        canvas_course_id: "$Canvas.course.id",
+        external_tool_url: "$Canvas.externalTool.url",
+      },
+      account_navigation: {
+        text: "Manage Users",
```
Comment (srphillips) [inline db/seeds.rb]: Do we have an official name for this tool so we can present it properly?

File: db/seeds.rb (hunk 1 above)
Comment (conradbeach) [reply db/seeds.rb]: Ok. I'll add `EDIT_USER_LOGIN` to cover editing logins.

Which endpoint would need to be added for searching for users? I'm not seeing one. `LIST_USERS_IN_ACCOUNT ` allows adding a `search_term` parameter, but it doesn't look like it supports all the fields we want to search on.

File: db/seeds.rb (hunk 4 above)
Comment (conradbeach) [reply db/seeds.rb]: Yeah, that'd be good. "Manage Users" just sounded like a reasonable temporary name to me.

File: app/models/application.rb (hunk 3 above)
Comment (bfcoder) [reply app/models/application.rb]: It does with the way the seeds set it up. Which is unfortunate. I don't want to put `viper` right here. This is the correct lti-key that should be used. So I'm thinking we will figure out a way to do this without hardcoding `viper` in here.

File: db/seeds.rb (hunk 1 above)
Comment (conradbeach) [reply db/seeds.rb]: Sorry, your second comment wasn't displaying at first for whatever reason.

Ok. I'll add `EDIT_USER_LOGIN`.

Comment (conradbeach) [reply db/seeds.rb]: Added

File: db/seeds.rb [hunk 5]
```diff
@@ -384,6 +384,45 @@
...
+        visibility: "admins",
+      },
+    },
+    application_instances: [{
+      lti_key: Application::USERTOOL,
+      site_url: secrets.canvas_url,
+      canvas_token: secrets.canvas_token,
+      domain: "#{Application::USERTOOL}.#{secrets.application_root_domain}",
```
Comment (bfcoder) [inline db/seeds.rb]: This is where we will have to decide how to populate this subdomain with `viper`. Be it hard coding or through the secrets and the cluster setup.

File: client/apps/user_tool/components/layout/index.spec.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (conradbeach) [reply client/apps/user_tool/components/layout/index.spec.jsx]: :+1:

File: db/seeds.rb (hunk 5 above)
Comment (conradbeach) [reply db/seeds.rb]: Ok. Do you want to do that now, or is it good as it is for the time being?

Comment (bfcoder) [reply db/seeds.rb]: Leave it for now, and we will modify it later.

Comment (conradbeach) [reply db/seeds.rb]: I confirmed with James that this is fine for now.

**Pull request merged**